SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -222,13 +223,91 @@ void CIncNSSolver::GetStreamwise_Periodic_Properties(const CGeometry *geometry,
           HeatFlow_Local += FaceArea * (-1.0) * Wall_HeatFlux/config->GetHeat_Flux_Ref();;
         } // loop Vertices
       } // loop Heatflux marker
+
+      if (config->GetMarker_All_KindBC(iMarker) == ISOTHERMAL) {
+        /*--- Identify the boundary by string name and retrive ISOTHERMAL from config ---*/
+
+        for (auto iVertex = 0ul; iVertex < geometry->nVertex[iMarker]; iVertex++) {
+
+          const auto iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
+
+          if (!geometry->nodes->GetDomain(iPoint)) continue;
+
+          /*--- Compute wall heat flux (normal to the wall) based on computed temperature gradient ---*/
+          const auto AreaNormal = geometry->vertex[iMarker][iVertex]->GetNormal();
+
+          su2double GradT[MAXNDIM] = {0.0,0.0,0.0};
+          for (auto iDim = 0u; iDim < nDim; iDim++)
```

Comment:
So I have to revert this as GetGradient_Primitive gives a Matrix su2double which does not comply with DotProduct. I saw at other parts of the code for loop being used to construct a variable and then do DotProduct. 

Do you know a better way to handle this?

Reply:
Sorry, slightly wrong suggestion, do this instead please:
```
const auto GradT = nodes->GetGradient_Primitive(iPoint)[prim_idx.Temperature()];
```